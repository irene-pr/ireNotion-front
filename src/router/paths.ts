const paths = {
  home: "/",
  login: "/login",
  register: "/register",
  userBoard: "/userBoard",
};

export default paths;
