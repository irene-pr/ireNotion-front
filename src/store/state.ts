const state = {
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
};

export default state;
