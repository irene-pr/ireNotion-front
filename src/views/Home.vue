<template>
  <main class="home-page">
    <div class="full-height-div full-height-div--top">
      <UserArea />
      <h1 class="home-logo">IRE<span>N</span>OTION</h1>
    </div>
    <section class="text-section text-section--light">
      <h2>About the website</h2>
      <p>
        This is my final project for the ISDI Coders fulltime software development bootcamp, I hope
        to show everything I've learned and that it doesn't just work in my
        computer. It's an app that registers users and offers them boards to
        store different sorts of data in form of post-its. The name ireNotion was born in the
        imagination of Pia Borges, there is no copyright claim because neither
        of us believes in it.
      </p>
    </section>
    <div class="medium-height-div medium-height-div--first"></div>
    <section class="text-section text-section--dark">
      <h2>Technologies I've used</h2>
      <p>
        It's a Full Stack development project written with Typescript. The
        Frontend it's a VUE SPA. In the Backend I've used the framework
        Express to make a REST API to store the users information and their data.
        Everything is tested with Jest and Cypress.
      </p>
    </section>
    <div class="medium-height-div medium-height-div--second"></div>
    <section class="text-section text-section--light">
      <h2>Message of Thanks</h2>
      <p>
        I thank all of my teachers and classmates for everything I've learned
        and experienced during the bootcamp. I've had the time of my life while
        suffering like never before. I also thank my mother for supporting me,
        I couldn't have done it without her.
      </p>
    </section>
    <div class="full-height-div full-height-div--bottom">
      <h1 class="home-logo">IRE<span>N</span>OTION</h1>
    </div>
  </main>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions, mapGetters, mapState } from "vuex";
import UserArea from "@/components/Home/NavHome.vue";

export default defineComponent({
  name: "Home",
  components: {
    UserArea,
  },
  computed: {
    ...mapState(["isLoggedIn"]),
  },
  methods: {
    ...mapActions(["checkToken"]),
    ...mapGetters(["redirectToUserBoard"]),
  },
  mounted() {
    this.checkToken();
    if (this.isLoggedIn) {
      this.redirectToUserBoard();
    }
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";

.home-page {
  font-family: $Lato;
  font-style: initial;
}
.full-height-div {
  height: 100vh;
  background: url("../assets/background-1.png");
  background-size: cover;
  background-attachment: fixed;

  .home-logo {
    margin: 0;
    width: fit-content;
    position: relative;
    bottom: calc(-50vh + 70px);
    left: calc(50vw - 230px);
    background-color: $theme-dark-color-opacity;
    color: $theme-light-color;
    padding: 20px;
    padding-left: 35px;
    font-size: 36px;
    font-weight: 100;
    letter-spacing: 0.6em;
    text-align: center;

    span {
      color: $theme-pink;
      font-weight: 300;
    }

    @media only screen and (max-width: 480px) {
      width: 100%;
      left: 0%;
      font-size: 24px;
    }
  }
}
.medium-height-div {
  height: 60vh;

  &--first {
    background: url("../assets/background-4.png");
    background-size: cover;
    background-attachment: fixed;
  }
  &--second {
    background: url("../assets/background-3.webp");
    background-size: cover;
    background-attachment: fixed;
  }
}
.text-section {
  padding: 20px 70px;
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  &--light {
    background-color: $theme-light-color;
    color: $theme-dark-color;
  }

  &--dark {
    background-color: $theme-dark-color;
    color: $theme-light-color;
    p {
      font-weight: 300;
    }
  }
  h2 {
    margin: 15px;
    font-family: $heading-home;
    font-weight: bold;
    font-size: 24px;
    letter-spacing: 0.17em;
    text-align: center;
  }
  p {
    margin: 15px;
    text-align: center;
    font-family: $Lato;
    font-size: 16px;
    line-height: 33px;
    letter-spacing: 0.1em;
  }

  @media only screen and (max-width: 480px) {
    padding: 20px 20px;
  }

  @media only screen and (max-width: 720px) {
    padding: 20px 35px;
  }
}
</style>
