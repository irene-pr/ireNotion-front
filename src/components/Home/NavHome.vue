<template>
  <nav class="nav__user-area">
    <router-link :to="paths.login">
      <button class="nav__button nav__button--login" disabled>
        LOG <span>IN</span>
      </button>
    </router-link>
    <router-link :to="paths.login">
      <button class="nav__button nav__button--signup">
        SIGN <span>UP</span>
      </button>
    </router-link>
  </nav>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import paths from "../../router/paths";

export default defineComponent({
  name: "UserArea",
  data() {
    return {
      paths,
    };
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
@import "@/styles/_extends";

.nav__user-area {
  display: flex;
  position: fixed;
  z-index: 2;
}
.nav__button {
  position: absolute;
  top: 14px;
  @extend %button;

  &--login {
    @extend %button--login;
    left: 14px;
  }
  &--signup {
    @extend %button--signup;
    left: 116px;
  }
  span {
    @extend %button--span;
    color: $theme-pink;
  }
}
</style>
