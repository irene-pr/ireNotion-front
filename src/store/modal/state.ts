import { IModalState } from "@/types/store";

const state = (): IModalState => ({
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
});

export default state;
