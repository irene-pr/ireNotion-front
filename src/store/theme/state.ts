import { IThemeState } from "@/types/store";

const state = (): IThemeState => ({
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
});

export default state;
