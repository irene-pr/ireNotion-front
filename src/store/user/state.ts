import { IUserState } from "@/types/store";

const state = (): IUserState => ({
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  isLoading: false,
});

export default state;
