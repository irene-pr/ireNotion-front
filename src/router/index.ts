import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import paths from "@/router/paths";
import Home from "../views/Home.vue";
import Login from "../views/Login.vue";
import Register from "../views/Register.vue";
import BoardPage from "../views/BoardPage.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: paths.home,
    name: "Home",
    component: Home,
  },
  {
    path: paths.login,
    name: "Login",
    component: Login,
  },
  {
    path: paths.register,
    name: "Register",
    component: Register,
  },
  {
    path: paths.userBoard,
    name: "BoardPage",
    component: BoardPage,
  },
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
