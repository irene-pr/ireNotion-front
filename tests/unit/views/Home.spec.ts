import { mount, shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
import Home from "../../../src/views/Home.vue";
import NavHome from "../../../src/components/Home/NavHome.vue";
import state from "@/store/state";
import router from "@/router";

describe("Given a Home view", () => {
  let checkTokenMock = jest.fn();
  let redirectToUserBoardMock = jest.fn();
  let store = createStore({});

  beforeEach(() => {
    checkTokenMock = jest.fn();
    redirectToUserBoardMock = jest.fn();
    store = createStore({
      getters: {
        redirectToUserBoard: redirectToUserBoardMock,
      },
      modules: {
        user: {
          namespaced: true,
          state() {
            return state;
          },
          actions: {
            checkToken: checkTokenMock,
          },
        },
      },
    });
  });

  describe("When it is rendered", () => {
    test("Then it renders", () => {
      mount(Home, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
    });
    test("Then the it matches the snapshot", () => {
      const wrapper = mount(Home, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Then the action checkToken should be called", () => {
      mount(Home, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      expect(checkTokenMock).toHaveBeenCalled();
    });
    test("Then the getter redirectToUserBoard should not be called", () => {
      mount(Home, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      expect(redirectToUserBoardMock).not.toHaveBeenCalled();
    });
    test("it renders the component NavHome", () => {
      const wrapper = mount(Home, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      wrapper.getComponent(NavHome);
    });
  });
  describe("When the isLoggedIn state is true", () => {
    test("Then it calls the getter redirectToUserBoard", () => {
      state.isLoggedIn = true;

      mount(Home, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-link", "router-view"],
      });
      expect(redirectToUserBoardMock).toHaveBeenCalled();
    });
    test("Then it redirects to /userBoard", () => {
      state.isLoggedIn = true;
      const $route = { path: "/userBoard" };

      const wrapper = mount(Home, {
        global: {
          plugins: [router, store],
          mocks: {
            $route,
          },
        },
        stubs: ["router-link", "router-view"],
      });
      expect(wrapper.vm.$route.path).toContain($route.path);
    });
  });
});
