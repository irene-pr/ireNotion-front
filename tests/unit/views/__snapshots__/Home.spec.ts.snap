// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Home view When it is rendered Then the it matches the snapshot 1`] = `
<main
  class="home-page"
>
  <div
    class="full-height-div full-height-div--top"
  >
    <nav
      class="nav__user-area"
    >
      <button
        class="nav__button nav__button--login"
      >
         LOG 
        <span>
          IN
        </span>
      </button>
      <button
        class="nav__button nav__button--signup"
      >
         SIGN 
        <span>
          UP
        </span>
      </button>
    </nav>
    <h1
      class="home-logo"
    >
      IRE
      <span>
        N
      </span>
      OTION
    </h1>
  </div>
  <section
    class="text-section text-section--light"
  >
    <h2>
      About the website
    </h2>
    <p>
       This is my final project for the ISDI Coders fulltime software development bootcamp, I hope to show everything I've learned and that it doesn't just work in my computer. It's an app that registers users and offers them boards to store different sorts of data in form of post-its. The name ireNotion was born in the imagination of Pia Borges, there is no copyright claim because neither of us believes in it. 
    </p>
  </section>
  <div
    class="medium-height-div medium-height-div--first"
  />
  <section
    class="text-section text-section--dark"
  >
    <h2>
      Technologies I've used
    </h2>
    <p>
       It's a Full Stack development project written with Typescript. The Frontend it's a VUE SPA. In the Backend I've used the framework Express to make a REST API to store the users information and their data. Everything is tested with Jest and Cypress. 
    </p>
  </section>
  <div
    class="medium-height-div medium-height-div--second"
  />
  <section
    class="text-section text-section--light"
  >
    <h2>
      Message of Thanks
    </h2>
    <p>
       I thank all of my teachers and classmates for everything I've learned and experienced during the bootcamp. I've had the time of my life while suffering like never before. I also thank my mother for supporting me, I couldn't have done it without her. 
    </p>
  </section>
  <div
    class="full-height-div full-height-div--bottom"
  >
    <h1
      class="home-logo"
    >
      IRE
      <span>
        N
      </span>
      OTION
    </h1>
  </div>
</main>
`;
